pkg/sql/opt_catalog.go

```diff
@@ -975,13 +975,18 @@ func newOptTable(
 
 		if idx.IsUnique() {
 			if idx.ImplicitPartitioningColumnCount() > 0 {
-				// Add unique constraints for implicitly partitioned unique indexes.
+				// Add unique constraints for implicitly partitioned unique indexes. If there is a single implicit column
+				// of ENUM type (e.g. an implicit RBR column), then we can ensure uniqueness under non-Serializable
+				// isolation levels by writing tombstones
+				canUseTombstones := idx.ImplicitPartitioningColumnCount() == 1 &&
+					catalog.FindColumnByID(desc, idx.GetKeyColumnID(0)).GetType().Family() == types.EnumFamily
```

Comment:
I'm uncomfortable assuming the RBR column is '0'. Is there a better way to figure this out? At the very least it feels like I should be using a named constant here.